net/dns/resolved.go

```diff
@@ -167,11 +168,30 @@ func (m *resolvedManager) SetDNS(config OSConfig) error {
 		})
 	}
 
-	err = m.resolved.CallWithContext(
-		ctx, "org.freedesktop.resolve1.Manager.SetLinkDomains", 0,
-		m.ifidx, linkDomains,
-	).Store()
-	if err != nil {
+	for try := 1; try < 2; try++ {
```

Comment:
Can you inline the two attempts rather than loop? I find the looping harder to follow than two calls separated by error handling logic.

Reply:
done